pa/src/components/CreateButton.vue

```diff
@@ -21,128 +21,128 @@
-        fin_api: function () {
-            return this.$store.getters.fin_api
-        },
-    },
-    created: function () {
-        this.getCreationOptions();
-    },
-    methods: {
-        getCreationOptions: function () {
-            this.fin_api
-                .options(this.apiUrl)
-                .then(response => {
-                    if (response.data) {
-                        let creationData = response.data['actions']['POST'];
-
-                        if (creationData['query_params']) {
-                            let query_params = creationData['query_params'];
-                            delete creationData['query_params'];
-                            for (const query_param in query_params) {
-                                if (query_params.hasOwnProperty(query_param) &&
-                                    !query_params[query_param]['read_only']) {
-                                    this.fieldsInfo[query_param] = query_params[query_param];
-                                    this.fieldsInfo[query_param]['query_param'] = true;
-                                }
-                            }
-                        }
+  },
+  created: function () {
+    this.getCreationOptions();
+  },
+  methods: {
+    getCreationOptions: function () {
+      this.fin_api
+        .options(this.apiUrl)
+        .then(response => {
+          if (response.data) {
+            let creationData = response.data['actions']['POST'];
 
-                        for (let field of Object.keys(creationData)) {
-                            if (creationData.hasOwnProperty(field) &&
-                                !creationData[field]['read_only']) {
-                                this.fieldsInfo[field] = creationData[field];
-                            }
-                        }
-                    }
-                })
-                .catch(errorResponse => {
-                    this.$bvToast.toast(`${errorResponse}`, this.errorMsgConfig)
-                })
+            if (creationData['query_params']) {
+              let query_params = creationData['query_params'];
+              delete creationData['query_params'];
+              for (const query_param in query_params) {
+                if (query_params.hasOwnProperty(query_param) &&
+                  !query_params[query_param]['read_only']) {
+                  this.fieldsInfo[query_param] = query_params[query_param];
+                  this.fieldsInfo[query_param]['query_param'] = true;
+                }
+              }
+            }
 
-        },
-        submitForm: function (bvModalEvt) {
-            bvModalEvt.preventDefault();
+            for (let field of Object.keys(creationData)) {
+              if (creationData.hasOwnProperty(field) &&
+                !creationData[field]['read_only']) {
+                this.fieldsInfo[field] = creationData[field];
+              }
+            }
+          }
+        })
+        .catch(errorResponse => {
```

Comment:
the above method and this one look too general you can extract them to a dedicated file and reuse the logic everywhere you need them. 
Also, such catch is handled nicely with axios interceptors

Reply:
the below method on submit the form also looks general and doesn't to a component.

it's a good practice to make component 'displayable' and extract 'logical' parts elsewhere